tests/rust-integration-tests/integration_test/src/tests/hostname/mod.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::{Context, Result};
+use oci_spec::runtime::{LinuxBuilder, ProcessBuilder, Spec, SpecBuilder};
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::utils::test_inside_container;
+
+fn create_spec(hostname: &str) -> Result<Spec> {
+    Ok(SpecBuilder::default()
+        .hostname(hostname)
+        .linux(
+            LinuxBuilder::default()
+                .readonly_paths(vec![
+                    "/proc/bus".to_string(),
+                    "/proc/fs".to_string(),
+                    "/proc/sys".to_string(),
+                ])
+                .build()
+                .expect("should build linux config"),
+        )
+        .process(
+            ProcessBuilder::default()
+                .args(vec!["runtimetest".to_string()])
+                .build()
+                .expect("can create process"),
+        )
+        .build()
+        .context("Failed to build hostname spec")?)
+}
+
+fn hostname_test() -> TestResult {
+    let cases = vec!["hostname-specific", ""];
+    for case in cases {
+        let spec = create_spec(case).expect("should create spec");
```

Comment:
```suggestion
        let spec = create_spec(case).expect("error in creating spec");
```
Alternatively unwrap in the `create_spec` as suggested in other comment